File: backend/src/main/java/com/backend/ord/repositories/UserSessionRepository.java
package com.backend.ord.repositories;

import com.backend.ord.domain.entities.UserSession;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface UserSessionRepository extends JpaRepository<UserSession, String> {
    Optional<UserSession> findByToken(String token);

    @Query("SELECT us FROM UserSession us WHERE us.token = :token AND us.user.id = :userId")
    Optional<UserSession> findByTokenAndUserId(String token, UUID userId);
}

File: backend/src/main/kotlin/com/backend/ord/repositories/UserSessionRepository.kt
package com.backend.ord.repositories

import com.backend.ord.domain.persistence.entities.UserSession
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserSessionRepository : JpaRepository<UserSession, String> {
    fun findByToken(token: String): UserSession?

    @Query("SELECT us FROM UserSession us WHERE us.token = :token AND us.user.id = :userId")
    fun findByTokenAndUserId(token: String, userId: UUID): UserSession?

    @Modifying
    @Query("DELETE FROM UserSession us WHERE us.token = :token")
    fun deleteByToken(token: String): Unit
}
